### PR TITLE
Add list services and add service feature

### PR DESCRIPTION
**Pull request by flairekq**

### CODE REVIEW

File: src/main/java/seedu/trackbeau/model/ReadOnlyTrackBeau.java [hunk 1]
```diff
@@ -14,4 +15,9 @@
      */
     ObservableList<Customer> getCustomerList();
 
+    /**
+     * Returns an unmodifiable view of the customers list.
+     * This list will not contain any duplicate customers.
+     */
+    ObservableList<Service> getServiceList();
```
Comment (kashing555) [inline src/main/java/seedu/trackbeau/model/ReadOnlyTrackBeau.java:22]: Should be servicelist instead

File: src/main/java/seedu/trackbeau/model/TrackBeau.java [hunk 2]
```diff
@@ -93,28 +106,74 @@ public void removeCustomer(Customer key) {
...
+    public boolean hasService(Service service) {
+        requireNonNull(service);
+        return services.contains(service);
+    }
+
+    /**
+     * Adds a customer to trackBeau.
+     * The customer must not already exist in trackBeau.
+     */
+    public void addService(Service s) {
```
Comment (kashing555) [inline src/main/java/seedu/trackbeau/model/TrackBeau.java:123]: Should be adding a service instead

**Review by kashing555 — APPROVED** [review]
LGTM

File: src/main/java/seedu/trackbeau/model/ReadOnlyTrackBeau.java (hunk 1 above)
Comment (flairekq) [reply src/main/java/seedu/trackbeau/model/ReadOnlyTrackBeau.java:22]: Updated

File: src/main/java/seedu/trackbeau/model/TrackBeau.java (hunk 2 above)
Comment (flairekq) [reply src/main/java/seedu/trackbeau/model/TrackBeau.java:123]: Updated

File: src/main/java/seedu/trackbeau/model/service/UniqueServiceList.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    /**
+     * Replaces the service {@code target} in the list with {@code editedService}.
+     * {@code target} must exist in the list.
+     * The service identity of {@code editedService} must not be the same as another existing service in the list.
+     */
+    public void setService(Service target, Service editedService) {
+        requireAllNonNull(target, editedService);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
```
Comment (isabelteo) [inline src/main/java/seedu/trackbeau/model/service/UniqueServiceList.java:60]: can consider setting INVALID_INDEX = -1?

**Review by isabelteo — COMMENTED** [review]
LGTM

File: src/main/java/seedu/trackbeau/model/service/UniqueServiceList.java (hunk 3 above)
Comment (flairekq) [reply src/main/java/seedu/trackbeau/model/service/UniqueServiceList.java:60]: Sounds good thanks!

**Pull request merged**